jetty-util/src/main/java/org/eclipse/jetty/util/Pool.java

```diff
@@ -559,4 +574,222 @@ public String toString()
                 pooled);
         }
     }
+
+    /** A pluggable strategy to optimize pool acquisition
+     * @param <T> The type of the items in the pool
+     */
+    public interface Cache<T>
+    {
+        /** Acquire an entry
+         * @param entries The list of entries known to the pool. This may be concurrently modified.
+         * @return An acquired entry or null if none can be acquired by this strategy
+         */
+        Pool<T>.Entry acquire(List<Pool<T>.Entry> entries);
+
+        /**
+         * Notification an entry has been release.  The notification comes after the entry
+         * has been put back in the pool and it may already have been reacquired before or during this call.
+         * @param entries The list of entries known to the pool. This may be concurrently modified.
+         * @param entry The entry to be release
+         * @param reusable true if the entry is reusable and will be put back in the pool.
+         */
+        default void released(List<Pool<T>.Entry> entries, Pool<T>.Entry entry, boolean reusable)
+        {
+        }
+    }
+
+    public interface Strategy<T> extends Cache<T>
```

Comment:
Are both `Strategy` and `Cache` needed?

Reply:
Strategy and Cache are needed if we want to make a distinction between impls that just try to find an entry and then give up (threadlocal, random) and those that do a search  ( linear, round-robin).

However, I too am not convinced the distinction is necessary.